### PR TITLE
feat: Query Import to support Oracle INTERVAL_YM

### PR DESCRIPTION
**Pull request by nj1973**
This is a code change only without tests, this is because it is a pre-req to merging PR for https://github.com/gluent/goe/issues/87 which implements a test.

### CODE REVIEW

File: src/goe/offload/offload_transport.py [hunk 1]
```diff
@@ -357,7 +377,10 @@ def log(msg, messages):
         return False
 
     if offload_source_table:
-        if ORACLE_TYPE_INTERVAL_YM in offload_source_table.data_types_in_use():
+        if (
+            ORACLE_TYPE_INTERVAL_YM in offload_source_table.data_types_in_use()
+            and 1 == 2
```
Comment (abb9979) [inline src/goe/offload/offload_transport.py:382]: Is this `1 == 2` deliberate?

File: src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py [hunk 2]
```diff
@@ -344,26 +346,25 @@ def get_rdbms_query_cast(
...
             and convert_expressions_on_rdbms_side
         ):
-            cast_expression = "TO_CHAR(%s,'TM')" % column_expression
+            cast_expression = f"TO_CHAR({column_expression},'TM')"
         elif (
             rdbms_column.data_type in (ORACLE_TYPE_INTERVAL_DS, ORACLE_TYPE_INTERVAL_YM)
             and for_spark
-        ):
-            cast_expression = "TO_CHAR(%s)" % column_expression
+        ) or (rdbms_column.data_type == ORACLE_TYPE_INTERVAL_YM and for_qi):
```
Comment (abb9979) [inline src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py:366]: If intervals are converted to string, why do we differentiate between supporting DS and YM for Spark and only YM for Query Import?

**Review by abb9979 — COMMENTED** [review]
A couple of inline comments in the code

File: src/goe/offload/offload_transport.py (hunk 1 above)
Comment (nj1973) [reply src/goe/offload/offload_transport.py:382]: No it isn't! How did I miss that! Thanks

File: src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py (hunk 2 above)
Comment (nj1973) [reply src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py:366]: It's based on what cx-Oracle supports. We get the exception below when trying to query interval YM:
```
    ora_cursor.execute(source_query)
cx_Oracle.NotSupportedError: Oracle type 2016 not supported.
```
But interval DS is supported and is currently offloaded. So I was just plugging the gap in cx-Oracle support.

File: src/goe/offload/offload_transport.py (hunk 1 above)
Comment (nj1973) [reply src/goe/offload/offload_transport.py]: I've removed this code.

File: src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py (hunk 2 above)
Comment (abb9979) [reply src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py:366]: We can simplify this to applying the TO_CHAR cast expression to DS and YM for both Spark and Qi.

Comment (nj1973) [reply src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py]: Agreed, I've made the change.

**Review by abb9979 — APPROVED** [review]

**Pull request merged**